packages/ssx-gnosis-extension/src/modal.ts

```diff
@@ -471,14 +471,15 @@ export class GnosisDelegation implements SSXExtension {
    * @returns Promise void.
    */
   connect = async (): Promise<void> => {
-    if (!this.selectedOption.replace(/Yourself - /, '')) {
+    const option = this.selectedOption.replace(/Yourself - /, '');
+    if (!option || [...delegators, this._connectedAddress].filter(delegator => option === delegator).length !== 1) {
```

Comment:
The visual comprehension here should be improved. Recommend assigning the result of that filter result/condition check to a variable and referencing the variable in the conditional block.